bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,473 @@
+                logger.warn("Message payload in binary format. Don't know how to handle it. Please report it.");
+                messageText = bytes.toString();
+            } else {
+                logger.error("Message has no payload !");
+                return;
+            }
+        }
+        logger.debug("Receiving new message from {} : {}", sender, messageText);
+
+        // dispatch to conversation :
+        for (SMSConversationHandler child : childHandlers) {
+            child.checkAndReceive(sender, messageText);
+        }
+
+        // channel trigger
+        String recipientAndMessage = sender + "|" + messageText;
+        triggerChannel(SMSModemBindingConstants.CHANNEL_TRIGGER_MODEM_RECEIVE, recipientAndMessage);
+
+        // prepare discovery service
+        senderMsisdn.add(sender);
+        final SMSConversationDiscoveryService finalDiscoveryService = discoveryService;
+        if (finalDiscoveryService != null) {
+            finalDiscoveryService.buildByAutoDiscovery(sender);
+        }
+        try { // delete message on the sim
+            modem.delete(message);
+        } catch (CommunicationException e) {
+            logger.error("Cannot delete message after receiving it !", e);
+        }
+    }
+
+    /**
+     * Send message
+     *
+     * @param recipient The recipient for the message
+     * @param text The message content
+     * @param deliveryReport If we should ask the network for a delivery report
+     */
+    public void send(String recipient, String text, boolean deliveryReport, @Nullable String encoding) {
+        OutboundMessage out = new OutboundMessage(recipient, text);
+        try {
+            if (encoding != null && !encoding.isEmpty()) {
+                Encoding encoding2 = Encoding.valueOf(encoding);
+                out.setEncoding(encoding2);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Encoding {} is not supported. Use Enc7, Enc8, EncUcs2, or EncCustom", encoding);
+        }
+        out.setRequestDeliveryReport(deliveryReport);
+        logger.debug("Sending message to {}", recipient);
+        modem.queue(out);
+    }
+
+    /**
+     * Used by the scanning discovery service to create conversation
+     *
+     * @return All senders of the received messages since the last start
+     */
+    public Set<String> getAllSender() {
+        return new HashSet<>(senderMsisdn);
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(SMSModemActions.class, SMSConversationDiscoveryService.class);
+    }
+
+    @Override
+    public boolean processStatusCallback(Modem.Status oldStatus, Modem.Status newStatus) {
+        switch (newStatus) {
+            case Error:
+                logger.error("SMSLib reported an error on the underlying modem {}", modem.getDescription());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
You could add the error message as a third argument to `updateStatus()`. Then, the error message is displayed in the UI and the logging could be removed.

Reply:
Thanks, done